deployments/helm/nvidia-dra-driver-gpu/templates/kubeletplugin.yaml

```diff
@@ -98,6 +98,19 @@ spec:
           compute-domain-kubelet-plugin -v 6
         resources:
           {{- toYaml .Values.kubeletPlugin.containers.computeDomains.resources | nindent 10 }}
+        {{/*
+          A literal "0" will allocate a random port. Don't configure the probe
```

Comment:
Hm. When instantiating a TCP listener, a literal 0 will lead to a system-provided port -- yes.

Here, we specify where to connect _to_ (if I am not mistaken), so specifying zero probably never makes quite sense. Happy to learn where I am off!

Reply:
The code below has this check:
```
{{- if (gt (int .Values.kubeletPlugin.containers.plugin.healthcheckPort) 0) }}
```
So if one was to specify 0 in the values.yaml file, then the section to specify the port below will be omitted.
Essentially not running a liveness probe at all.

A better implementation would probably have been to put some code in `validation.yaml` to error out if `healthcheckPort == 0` rather than silently omitting the liveness probe altogether.

Let's leave that for a follow-up so that this code continues to follow the original PR from the example driver as closely as possible.